code/game/objects/structures/window.dm

```diff
@@ -276,7 +264,26 @@
 		return
 	playsound(src, "shatter", 70, 1)
 	var/turf/T = loc
-	for(var/obj/item/I in storeditems)
+	var/produced_items = list()
+
+	if(!(flags & NODECONSTRUCT))
+		var/shards = 1
+		if(fulltile)
+			shards++
+		var/rods = 0
+		if(reinf)
+			rods++
+			if(fulltile)
+				rods++
+
+		for(var/i in 1 to shards)
+			produced_items += new /obj/item/weapon/shard(src)
+		if(rods)
+			produced_items += new /obj/item/stack/rods(src, rods)
```

Comment:
If this is something we should be adopting for the majority of popular structure (walls, windows, floors), should we make some sort of var maybe at /obj/structure level for storing components and release them under certain conditions?